doc/custom-metadata-feature.md

```diff
@@ -0,0 +1,191 @@
+            "objectValue" : {
+               "Color" : "Black and white",
+               "CommonName" : "Front Door Camera",
+               "Condition" : "Good working condition",
+               "Location" : "Front Door"
+            },
+            "origin" : 1655409814077374935,
+            "profileName" : "onvif-camera",
+            "resourceName" : "CustomMetadata",
+            "value" : "",
+            "valueType" : "Object"
+         }
+      ],
+      "sourceName" : "CustomMetadata"
+   },
+   "statusCode" : 200
+}
+```
+
+
+
+### Get Specific Custom Metadata
+
+Pass the *CustomMetadata* resource a query to get specific field(s) in CustomMetadata. The query must be a base64 encoded json object with an array of fields you want to access.
+
+1. Json object holding an array of fields you want to query.
+```json
+'{
+    "CustomMetadata": 
+        [
+            "CommonName",
+            "Location"
+        ]
+}'
+```
+
+2. Use this command to convert the json object to base64.
+```shell
+echo '{
+    "CustomMetadata": 
+        [
+            "CommonName",
+            "Location"
+        ]
+}' | base64
+```
+
+3. The response converted to base64.
+```shell
+ewogICAgIkN1c3RvbU1ldGFkYXRhIjogCiAgICAgICAgWwogICAgICAgICAgICAiQ29tbW9uTmFtZSIsCiAgICAgICAgICAgICJMb2NhdGlvbiIKICAgICAgICBdCn0K
+```
+
+4. Use this command to query the fields you provided in the json object.
+```shell
+curl http://localhost:59882/api/v2/device/name/3fa1fe68-b915-4053-a3e1-cc32e5000688/CustomMetadata?jsonObject=ewogICAgIkN1c3RvbU1ldGFkYXRhIjogCiAgICAgICAgWwogICAgICAgICAgICAiQ29tbW9uTmFtZSIsCiAgICAgICAgICAgICJMb2NhdGlvbiIKICAgICAgICBdCn0K | json_pp
+
+```
+
+5. Curl response. 
+```shell
+{
+   "apiVersion" : "v2",
+   "event" : {
+      "apiVersion" : "v2",
+      "deviceName" : "3fa1fe68-b915-4053-a3e1-cc32e5000688",
+      "id" : "24c3eb0a-48b1-4afe-b874-965aeb2e42a2",
+      "origin" : 1655410556448058195,
+      "profileName" : "onvif-camera",
+      "readings" : [
+         {
+            "deviceName" : "3fa1fe68-b915-4053-a3e1-cc32e5000688",
+            "id" : "d0c26303-20b5-4ccd-9e63-fb02b87b8ebc",
+            "objectValue" : {
+               "CommonName" : "Front Door Camera",
+               "Location" : "Front Door"
+            },
+            "origin" : 1655410556448058195,
+            "profileName" : "onvif-camera",
+            "resourceName" : "CustomMetadata",
+            "value" : "",
+            "valueType" : "Object"
+         }
+      ],
+      "sourceName" : "CustomMetadata"
+   },
+   "statusCode" : 200
+}
+```
+
+### Additional Usage
+
+Setting a field to "deletes" will delete that field.
+
+1. Use this command to delete fields.
+```shell
+curl --request PUT 'http://0.0.0.0:59882/api/v2/device/name/<device name>/CustomMetadata' \
+    --header 'Content-Type: application/json' \
+    --data-raw '{
+        "CustomMetadata": {
+            "Color":"delete",
+            "Condition": "delete"
+        }
+    }' | json_pp
+```
```

Comment:
I think we should find a better way to handle this. Each of them have some drawbacks or hackiness to them, but here are some alternatives:
1. Having a write-only `DeleteCustomMetadata` resource that takes the fields you want to delete as the json body
     1. Problem with this is that this `DeleteCustomMetadata` "resource" isnt really a "resource " so to speak.

2. Modifying the input object to have a special `DeleteMetadata` field (or other name) that you pass in a list of fields to delete.
     1. Depending on how it is possible may have to use csv string instead of array

3.  Keep it the same, but make the `delete` keyword less prone to collisions? `_DELETE_`? or something like that?

@trleasher-intel @lenny-intel thoughts?